compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -299,14 +300,31 @@ private void replaceMockedFunctions(BIRPackage birPkg, Map<String, String> mockF
                 if (bbTerminator.kind.equals(InstructionKind.CALL)) {
                     //We get the callee and the name and generate 'calleepackage#name'
                     BIRTerminator.Call callTerminator = (BIRTerminator.Call) bbTerminator;
-                    String functionKey = callTerminator.calleePkg.toString() + MOCK_ANNOTATION_DELIMITER
-                            + callTerminator.name.toString();
-                    if (mockFunctionMap.get(functionKey) != null) {
-                        // Just "get" the reference. If this doesnt work then it doesnt exist
-                        String mockfunctionName = mockFunctionMap.get(functionKey);
-                        callTerminator.name = new Name(mockfunctionName);
-                        callTerminator.calleePkg = function.pos.src.pkgID;
+
+                    // This will be an object function
+                    if (callTerminator.isVirtual && callTerminator.args.size() == 1) {
```

Comment:
Is this a viable way of identifying an object function? Are there better alternatives?

Either way, if it is not an object function, the generated `objectKey` will not be able to pull the relevant mock function from the mockFunctionMap and the program flow will continue without any changes.